### ISSUE TITLE
Null flow analysis

### ISSUE THREAD

**Issue by juliusfriedman** [dotnet/roslyn-analyzers#3848]
Is broken (not working) when used in string interpolation:

```c#
//This should be forced or something should be underlined because it could be null
if (somethingNullable != null) sb.Append(FormattableString.Invariant($@"""something"":""{somethingNullable}"","));
```

See also https://github.com/dotnet/roslyn-analyzers/issues/3834#issuecomment-655695890

**Comment by mavasani** [comment]
@juliusfriedman Can you clarify what null flow analysis are you referring to? Is this the C# 8 nullable reference type warnings? If so, this issue should be ported to roslyn repo.

**Comment by juliusfriedman** [comment]
I am not using nullable references so whatever null flow analysis is built into the analyzer that says to check for null.

**Comment by mavasani** [comment]
The only nullable related analyzer in this repo is CA1062, which only validates null check is performed for argument values of public APIs before first read. You will need to provide a much more clear repro case with actual and expected behavior for making this issue actionable.

**Comment by juliusfriedman** [comment]
Should it not also flow through to Actions, Lambdas etc or static functions declared in public methods?
It seems like a logic flow to follow to me but maybe we see things differently...

**Comment by sharwell** [comment]
@juliusfriedman it's not clear from the original post what the problem is. Can you update the first post to include a description of the **Expected behavior** and **Actual behavior**?
